check_limits.py

```diff
@@ -1,18 +1,42 @@
-
-def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
+def temp_is_ok(temperature):
+  if(temperature < 0 or temperature > 45):
+     print_console("temperature")
+     return False
+  return True
+    
+def soc_is_ok(soc):
+  if(soc <20 or soc > 80):
```

Comment:
The range comparisons are repeated for each parameter. Any enhancements there will need repeat-modifications. Try an `isInRange` function to isolate the comparisons

Reply:
Hello @sudeeprp ,

Modified the code based on your review points. As i need to modify the same code to complete Extend and Refactor assignment, kindly review the code and let me know the review points.